File: webant-commons/src/main/java/org/webant/commons/entity/HttpDataEntity.scala
package org.webant.commons.entity

import java.util.Date

import org.webant.commons.annotation.{DocId, DocParentId}

import scala.beans.BeanProperty

class HttpDataEntity extends Serializable {
  @DocId
  @DocParentId("parentId")
  @BeanProperty
  var id: String = _
  @BeanProperty
  var source: String = _
  @BeanProperty
  var srcId: String = _
  @BeanProperty
  var srcUrl: String = _
  @BeanProperty
  var taskId: String = _
  @BeanProperty
  var siteId: String = _
  @BeanProperty
  var crawlTime: Date = _
  @BeanProperty
  var dataVersion: java.lang.Integer = 1
  @BeanProperty
  var dataCreateTime: Date = new Date()
  @BeanProperty
  var dataUpdateTime: Date = new Date()
  @BeanProperty
  var dataDeleteTime: Date = _
}

File: webant-commons/src/main/java/org/webant/commons/entity/HttpDataEntity.java
package org.webant.commons.entity;

import org.webant.commons.annotation.DocId;
import org.webant.commons.annotation.DocParentId;

import java.io.Serializable;
import java.util.Date;

public class HttpDataEntity implements Serializable{
    @DocId
    @DocParentId("parentId")
    public String id;
    public String source;
    public String srcId;
    public String srcUrl;
    public String taskId;
    public String siteId;
    public Date crawlTime;
    public Integer dataVersion;
    public Date dataCreateTime;
    public Date dataUpdateTime;
    public Date dataDeleteTime;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getSource() {
        return source;
    }

    public void setSource(String source) {
        this.source = source;
    }

    public String getSrcId() {
        return srcId;
    }

    public void setSrcId(String srcId) {
        this.srcId = srcId;
    }

    public String getSrcUrl() {
        return srcUrl;
    }

    public void setSrcUrl(String srcUrl) {
        this.srcUrl = srcUrl;
    }

    public String getTaskId() {
        return taskId;
    }

    public void setTaskId(String taskId) {
        this.taskId = taskId;
    }

    public String getSiteId() {
        return siteId;
    }

    public void setSiteId(String siteId) {
        this.siteId = siteId;
    }

    public Date getCrawlTime() {
        return crawlTime;
    }

    public void setCrawlTime(Date crawlTime) {
        this.crawlTime = crawlTime;
    }

    public Integer getDataVersion() {
        return dataVersion;
    }

    public void setDataVersion(Integer dataVersion) {
        this.dataVersion = dataVersion;
    }

    public Date getDataCreateTime() {
        return dataCreateTime;
    }

    public void setDataCreateTime(Date dataCreateTime) {
        this.dataCreateTime = dataCreateTime;
    }

    public Date getDataUpdateTime() {
        return dataUpdateTime;
    }

    public void setDataUpdateTime(Date dataUpdateTime) {
        this.dataUpdateTime = dataUpdateTime;
    }

    public Date getDataDeleteTime() {
        return dataDeleteTime;
    }

    public void setDataDeleteTime(Date dataDeleteTime) {
        this.dataDeleteTime = dataDeleteTime;
    }
}
